realtime/setupSocketIO.js

```diff
@@ -116,7 +117,17 @@ function init(io, redisStore) {
           throw new Error(DISCO_MSG);
         }
 
-        rtUtils.isIpWhitelisted(ipAddress, ipWhitelist); // throws error
+        if (conf.ipWhitelistService) {
+          ipWhitelistUtils.isWhitelisted(ipAddress)
+            .then((allowed) => {
+              if (!allowed) {
+                socket.disconnect();
+                return;
```

Comment:
Any reason for doing this here instead of throwing an error to be handled later like we do currently?